lib/modules/manager/npm/post-update/pnpm.ts

```diff
@@ -41,9 +41,10 @@ export async function generateLockFile(
     const pnpmToolConstraint: ToolConstraint = {
       toolName: 'pnpm',
       constraint:
-        getPnpmConstraintFromUpgrades(upgrades) ??
-        config.constraints?.pnpm ??
-        (await getPnpmConstraint(lockFileDir)),
+        getPnpmConstraintFromUpgrades(upgrades) ?? // if pnpm is being upgraded, it comes first
+        config.constraints?.pnpm ?? // from user config or extraction
```

Comment:
on other managers we always prefer user config first. 🤔

Reply:
I think that's a mistake if we're explicitly upgrading `pnpm` in the PR. Also it won't work for pnpm because `constraints` includes extracted constraints from the `package.json` which will then be outdated. It would potentially be ok if `constraints` were only user-configured constraints but in this case it's not

Reply:
ok, so this works on pnpm update, but will fail all following updates when merged to main, because the user constraints are probably not yet updated.